models/model.js

```diff
@@ -6,17 +6,32 @@ exports.fetchTopics = () => {
     });
 };
 
-exports.fetchArticles = (id) => {
+exports.fetchArticleByID = (id) => {
     if (!/^[^A-z]+$/.test(id)) {
         return Promise.reject({ status: 400, message: 'Bad request! Enter a valid ID' })
     } else {
         return db.query(`SELECT * FROM articles 
         WHERE article_id = $1;`, [id])
-        .then((articleData) => {
-            if (articleData.rows.length !== 1) {
-                return Promise.reject({ status: 404, message: 'Article does not exist!' })
-            }
-            return articleData.rows[0]
-        })
+            .then((articleData) => {
+                if (articleData.rows.length !== 1) {
+                    return Promise.reject({ status: 404, message: 'Article does not exist!' })
+                }
+                return articleData.rows[0]
+            })
     }
-}
\ No newline at end of file
+}
+
+exports.fetchArticles = () => {
+    return db.query(`SELECT articles.*, COUNT(*)::INT AS comment_count FROM articles
+        JOIN comments ON articles.article_id = comments.article_id
+        GROUP BY articles.article_id ORDER BY articles.created_at DESC;`).then((data) => {
+        if (data.rows.length === 0) {
+            return Promise.reject({ status: 204, message: 'No Content!' })
+        } else {
+            return data.rows.map(article => {
```

Comment:
Nice work here with the data manipulation, however it would be better to handle this with SQL to only get back the columns we want rather than getting everything then deleting what we don't want.